subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/FileCategorizer.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.changedetection.state;
+
+public interface FileCategorizer {
```

Comment:
Looking for a better name.

Reply:
How about `ImmutableCachesRegistry`?

Reply:
It's not just caches though. Maybe "WellKnownFileLocations"

Reply:
For my $0.02, `WellKnownFileLocations` gives a lot more information about the purpose of this class than either `FileCategorizer` or `ImmutableCachesRegistry`.

Reply:
IIUC, all those files are in `CachedJarFileStore`s. Thats why I went for caches...

How about `ImmutableFileStoreLocations`?